components/tag-connect/TC2050-IDC-NL.stanza

```diff
@@ -12,26 +12,60 @@ defpackage ocdb/tag-connect/TC2050-IDC-NL :
   import ocdb/bundles
   import ocdb/generator-utils
   import ocdb/generic-components
+  import ocdb/box-symbol
+
+; NOTE UNUSUAL PINOUT
+; Datasheet showing pinout: https://www.tag-connect.com/wp-content/uploads/bsk-pdf-manager/TC2050-IDC-NL_Datasheet_8.pdf
 
 pcb-landpattern footprint :
-  for (l in grid-locs(2, 5, 1.27, 1.27), i in [1 3 5 7 9 2 4 6 8 10]) do :
+  for (l in grid-locs(2, 5, 1.27, 1.27), i in [1 2 3 4 5 10 9 8 7 6]) do :
     pad p[i] : testpoint-pad(0.787) at l
-  
   layer(Cutout()) = Circle((- 3.81), 0.0, 0.5)
   layer(Cutout()) = Circle(3.81, 1.015, 0.5)
   layer(Cutout()) = Circle(3.81, (- 1.015), 0.5)
   layer(Courtyard(Top)) = Rectangle(8.62, 3.03)
-  layer(Silkscreen("f-silk", Top)) = LineRectangle(8.62, 3.03)
   ref-label()
 
 public pcb-component component :
-  description = "Insertion point for TC-2050 connector"
+  description = "Insertion point for TC-2050-NL connector"
   port p: pin[1 through 10]
-  val sym = header-symbol(10,2)
-  symbol = sym(for i in 1 through 10 do : p[i] => sym.p[i])
+  pin-properties :
+    [pin:Ref     | pads:Int ... | side:Dir]
+    [p[1]          | 1             | Left     ]
+    [p[2]          | 2             | Left     ]
+    [p[3]          | 3             | Left     ]
+    [p[4]          | 4             | Left     ]
+    [p[5]          | 5             | Left     ]
+    [p[10]         | 10            | Right    ]
+    [p[9]          | 9             | Right    ]
+    [p[8]          | 8             | Right    ]
+    [p[7]          | 7             | Right    ]
+    [p[6]          | 6             | Right    ]
   landpattern = footprint(for i in 1 through 10 do : p[i] => footprint.p[i])
   reference-prefix = "J"
+  make-box-symbol()
+
+  supports jtag() :
+    jtag().tms => self.p[2]
+    jtag().tck => self.p[4]
+    jtag().tdo => self.p[6]
+    jtag().tdi => self.p[8]
+    jtag().trstn => self.p[10]
 
+  supports swd([SWD-SWO, SWD-TRACESWO]) :
+    swd([SWD-SWO, SWD-TRACESWO]).swdio  => self.p[2]
+    swd([SWD-SWO, SWD-TRACESWO]).swdclk => self.p[4]
+    swd([SWD-SWO, SWD-TRACESWO]).swo => self.p[6]
+    swd([SWD-SWO, SWD-TRACESWO]).traceswo => self.p[8]
+
+  supports power:
+    power.vdd => self.p[1]
+    power.gnd => self.p[3]
+    ; power.gnd => self.p[5]
+    ; power.gnd => self.p[7]
+    ; power.gnd => self.p[9]
+    ; net (power.gnd self.p[5])
```

Comment:
Do you want to keep these lines?

Reply:
yes, current;ly we can't do this. I would like this feature added. This also gives a heads up to folks using this to connect these pins IF they use the arm jtag